### PR TITLE
Fix spec

### LINKED ISSUE #2: Fix spec text
Current spec text uses the `finally` callback's return value as the promise resolution value (per https://github.com/ljharb/proposal-promise-finally/pull/1#issuecomment-219823021).

This needs to be fixed prior to stage 2.

### PR DESCRIPTION
**Pull request by ljharb**
Attempt to fix #2 and close #1.

Spec HTML viewable [here](https://rawgit.com/ljharb/proposal-promise-finally/fix_spec/index.html)

This attempts to fix the spec by:
1) spec-refactor of Promise.resolve logic so the meat of it is available as an abstract operation
2) use that abstract operation to produce a promise with certainty inside PromiseReactionJob, and resolve that with a thunk for the intended "argument".

### CODE REVIEW

**Comment by ljharb** [comment]
@domenic any thoughts on this solution?

File: spec.emu [hunk 1]
```diff
@@ -142,14 +170,18 @@ contributors: Jordan Harband
 			1. Else if _type_ is `"Reject"`, let _handlerResult_ be Completion{[[Type]]: ~throw~, [[Value]]: _argument_, [[Target]]: ~empty~}.
 			1. Else if _type_ is `"Finally"`, let _handlerResult_ be NormalCompletion(_argument_).
 			1. Assert: _handlerResult_ is defined.
-		1. Else if _type_ is `"Finally"`, let _handlerResult_ be Call(_handler_, *undefined*, &laquo; &raquo;).
+		1. Else if _type_ is `"Finally"`, then
+			1. let _handlerResult_ be Call(_handler_, *undefined*, &laquo; &raquo;).
+			1. If _handlerResult_ is not an abrupt completion, then
+				1. Let _handlerPromise_ be ? PromiseResolve(%Promise%, _handlerResult_.[[Value]]).
+				1. Let _argumentThunk_ be equivalent to a function that returns _argument_.
```
Comment (stefanpenner) [inline spec.emu]: neat, but can this be described like this without it being considered a layering violation? (I honestly don't know) @domenic ?

Comment (ljharb) [inline spec.emu]: This text is identical to text that used to be in the spec prior to https://github.com/tc39/ecma262/pull/584 so I'm pretty sure it's OK. Not sure what you mean by "layering violation" tho?

Comment (stefanpenner) [inline spec.emu]: Seems good then

File: index.html [hunk 2]
```diff
@@ -356,7 +356,7 @@ <h1 class="title">Promise.prototype.finally</h1>
 		</emu-alg>
 	</emu-clause>
 
-	<!-- es6num="25.4.1.8" -->
+	<!-- es6num="25.4.4.5" -->
```
Comment (ljharb) [inline index.html]: @bterlson this seems to have changed between when I initially wrote this PR and now. Was this intentional? I thought "es6num" values were supposed to stay constant forever?

Comment (bterlson) [inline index.html]: It's 25.4.4.5 in the spec now, so presumably this change is fine. Not sure where 25.4.1.8 came from. It should remain unchanged from ES6.

Comment (ljharb) [inline index.html]: Unless I copy-paste-errored, 25.4.1.8 is what it was when I first copied the spec steps for `Promise.resolve` and started writing this PR. Hopefully it's unchanged and I just messed it up :-)

**Pull request merged**